.github/workflows/csqa.yml

```diff
@@ -105,3 +105,37 @@ jobs:
 
       - name: Run Static Analysis
         run: composer static-analysis
+
+  perf-guard:
+    name: "Basic Performance Guard"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: "8.1"
+          coverage: none
+
+      # Install dependencies and handle caching in one go.
+      # Dependencies need to be installed to make sure the PHPUnit classes are recognized.
+      # @link https://github.com/marketplace/actions/install-php-dependencies-with-composer
+      - name: Install Composer dependencies
+        uses: "ramsey/composer-install@v3"
+        with:
+          # Bust the cache at least once a month - output format: YYYY-MM.
+          custom-cache-suffix: $(date -u "+%Y-%m")
+
+      - name: Download performance test fixture
+        run: wget https://raw.githubusercontent.com/WordPress/WordPress/refs/heads/master/wp-includes/PHPMailer/PHPMailer.php
+
+      - name: Run performance report
+        id: performance_report
+        run: echo "PHPCS_PERF_SECS=$(./vendor/bin/phpcs --standard=VariableAnalysis --report=Performance ./PHPMailer.php | grep -Eo 'TOTAL RUN TIME[ ]+[0-9.]+'|awk '{ print $4 }')" >> $GITHUB_OUTPUT
```

Comment:
I don't think using `TOTAL RUN TIME` is a good idea as it is also influenced by the time PHPCS itself takes, which may fluctuate if PHPCS would introduce a performance issue.

Most reliable should be to use the sniff specific line. Second most reliable, the `TOTAL SNIFF PROCESSING TIME` line.

Reply:
Good idea.